src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -319,5 +364,56 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
         {
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
+
+        private async Task GetImageNames(List<AugmentedTimelineRecord> records, CancellationToken cancellationToken)
+        {
+            SemaphoreSlim throttleSemaphore = new SemaphoreSlim(50);
+
+            var bag = new ConcurrentBag<Task>();
+            int canceledTasksNumber = 0;
+
+            foreach(var record in records)
+            {
+                if(!string.IsNullOrEmpty(record.Raw.Log?.Url) && record.Raw.Name == "Initialize job")
+                {
+                    var childTask = Task.Run(async () =>
+                    {
+                        try
+                        {
+                            await throttleSemaphore.WaitAsync();
+
+                            if (!cancellationToken.IsCancellationRequested)
+                            {
+                                record.ImageName = (record.Raw.WorkerName.StartsWith("Azure Pipelines") || record.Raw.WorkerName.StartsWith("Hosted Agent"))
+                                ? await _buildLogScraper.ExtractMicrosoftHostedPoolImageNameAsync(record.Raw.Log.Url, cancellationToken)
+                                : record.Raw.WorkerName.StartsWith("NetCore1ESPool-")
+                                    ? await _buildLogScraper.ExtractOneESHostedPoolImageNameAsync(record.Raw.Log.Url, cancellationToken)
+                                    : null;
+                            }
+                            else
+                            {
+                                Interlocked.Increment(ref canceledTasksNumber);
+                            }
+                        }
+                        catch (OperationCanceledException) 
+                        {
+                            Interlocked.Increment(ref canceledTasksNumber);
+                        }
+                        catch (Exception exception)
+                        {
+                            _logger.LogInformation($"Non critical exception thrown when trying to get log '{record.Raw.Log.Url}': {exception}");
```

Comment:
What reason do we have to belive this is a "non-critical exception"?  It's just a generic exception message.

Reply:
Non critical in the sense that we don't mind if some of the data doesn't get collected